### DISCUSSION TITLE
New to the library: trying to make a semicircle

### DISCUSSION THREAD

**Discussion by derekantrican** [Irev-Dev/Round-Anything#35] [Q&A] [answered]
I'm new to this library and trying to get set up with a simple semicircle first (part of a more complex shape I'm making). But I notice that this code:

```
//radius: 25
polygon(polyRound([
    [0, 25, 0],
    [25, 0, 25],
    [0, -25, 0],
],25));
```

Doesn't make a semicircle (see below image). Is there something I'm not understanding here?

![image](https://user-images.githubusercontent.com/1558019/142530944-4da44513-f3ad-4d2a-ba2a-116394222fda.png)

**Comment by derekantrican** [comment, marked as answer]
Ok, I had a "brainspark" and was able to get it with more points:

```
polygon(polyRound([
    [0, 25, 0],
    [25, 25, 25],
    [25, 0, 25],
    [25, -25, 25],
    [0, -25, 0],
],25));
```

I think maybe before it was because part of the semicircle would extend outside of the "non-rounded triangle"?

For instance, I think this is what I was trying to do before (triangle is non-rounded, semicircle is what I wanted):
![image](https://user-images.githubusercontent.com/1558019/142542552-bc60671b-db87-4266-baee-1a3ec30be72e.png)


And this is what the solution was (square is non-rounded, semicircle is what I wanted):
![image](https://user-images.githubusercontent.com/1558019/142542792-582cc87f-f9f1-4825-9677-9dc1b7a47252.png)


So the semicircle simply can't fit inside the triangle and therefore isn't really a possible "fillet" for that point

**Reply by Irev-Dev** [reply]
Hey @derekantrican,

Glad you figured out. I think it comes down a mental model of how it works. The library does not smooth a line that passes through a number of points. it instead fillets vertices. In my opinion is more useful as for CAD usually the positioning of the faces is more important and then being able to fillet the edges after is useful. 

You might consider [BOSL](https://github.com/revarbat/BOSL2) for spline utils.
